homeassistant/components/amberelectric/sensor.py

```diff
@@ -0,0 +1,591 @@
+                        self.data_service,
+                        self.data_service.coordinator,
+                    )
+                )
+
+                sensors.append(
+                    AmberForecastSensor(
+                        self._platform_name,
+                        self._site_id,
+                        ChannelType.FEED_IN,
+                        self.data_service,
+                        self.data_service.coordinator,
+                    )
+                )
+
+            if (
+                len(
+                    list(
+                        filter(
+                            lambda channel: channel.type == ChannelType.CONTROLLED_LOAD,
+                            self.data_service.site.channels,
+                        )
+                    )
+                )
+                > 0
+            ):
+                sensors.append(
+                    AmberPriceSensor(
+                        self._platform_name,
+                        self._site_id,
+                        ChannelType.CONTROLLED_LOAD,
+                        self.data_service,
+                        self.data_service.coordinator,
+                    )
+                )
+
+                sensors.append(
+                    AmberEnergyPriceSensor(
+                        self._platform_name,
+                        self._site_id,
+                        ChannelType.CONTROLLED_LOAD,
+                        self.data_service,
+                        self.data_service.coordinator,
+                    )
+                )
+
+                sensors.append(
+                    AmberForecastSensor(
+                        self._platform_name,
+                        self._site_id,
+                        ChannelType.CONTROLLED_LOAD,
+                        self.data_service,
+                        self.data_service.coordinator,
+                    )
+                )
+
+            sensors.append(
+                AmberRenewablesSensor(
+                    self._platform_name,
+                    self._site_id,
+                    self.data_service,
+                    self.data_service.coordinator,
+                )
+            )
+
+            sensors.append(
+                AmberPriceSpikeSensor(
+                    self._platform_name,
+                    self._site_id,
+                    self.data_service,
+                    self.data_service.coordinator,
+                )
+            )
+
+            LOGGER.debug("Adding %s sensors", str(len(sensors)))
+        else:
+            LOGGER.error("No site found!")
+        return sensors
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Amber integration."""
+    configuration = amberelectric.Configuration(
+        access_token=entry.data.get(CONF_API_TOKEN)
+    )
+
+    api_instance = amber_api.AmberApi.create(configuration)
```

Comment:
Why are we creating this again? Initialize the update coordinator in `__init__.py` and only forward to platforms if update was ok.